### PR TITLE
[SYS-742]: Add main event loop

### PR DESCRIPTION
**Pull request by akatch**
mqp will publish and process 5 messages, then exit.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,15 +27,69 @@ type MQ struct {
 
 // NewMQ initializes RabbitMQ and returns a pointer to it
 func NewMQ(uri *amqp.URI) (*MQ, error) {
+	log.Printf("connecting to %s", uri)
```
Comment (douglb) [inline main.go]: You are potentially leaking clear text passwords into your logs here if they are present in the URI.

File: main.go [hunk 2]
```diff
@@ -25,15 +27,69 @@ type MQ struct {
...
+func publishMessages(channel *amqp.Channel, msg string, expectedMessages int) {
+	for i := 1; i <= expectedMessages; i++ {
+		msg := fmt.Sprintf("%s: %d of %d", msg, i, expectedMessages)
+		publishMessage(channel, msg)
+		log.Printf("published: %s", msg)
+	}
+}
+
+func channelTimeout(channel *amqp.Channel, timeoutSeconds time.Duration) {
+	time.Sleep(timeoutSeconds * time.Second)
```
Comment (douglb) [inline main.go]: timeoutSeconds should be declared as an `int` and not `time.Duration`?  It looks like you are passing an int in for the value and not an actual duration.

File: main_test.go [hunk 3]
```diff
@@ -103,6 +90,11 @@ func (suite *MQPTestSuite) TestParseFlags() {
 	assert.Equal(suite.T(), want, got)
 }
 
+func (suite *MQPTestSuite) TestMain() {
```
Comment (douglb) [inline main_test.go]: I'm unclear on what is being tested here?

**Comment by akatch** [comment]
I believe I've addressed all of the previous feedback, please let me know if I missed anything.

**Pull request merged**